torch/csrc/jit/ivalue.h

```diff
@@ -91,10 +93,16 @@ using DoubleList = ConstantList<double>;
 #define TORCH_FORALL_TAGS(_) \
   _(None) _(Tensor) _(Double) _(Int) _(Tuple) _(IntList) _(DoubleList)
 
+enum class IValueKind : uint32_t {
```

Comment:
I strongly do not want to expose type tags of IValue as a public interface. It prevents future optimizations to the representation of the tags internally to the IValue. For instance, if we go to nan-boxing of 8-byte values then it is significantly easier to check `isTensor()` than to ask "of all the types what type is it?" Exposing the tags will encourage people to write `kind() == Tensor` and prevent future optimizations.

I don't thing anything in this PR actually requires this change. The `kind` field in ArgumentInfo change be change to an `is_tensor` field.

Reply:
Fair enough. I'll revert it.